pkg/logger/server.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2019 kubeflow.org.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logger
+
+import (
+	"bytes"
+	"github.com/go-logr/logr"
+	guuid "github.com/google/uuid"
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+type loggerHandler struct {
+	log       logr.Logger
+	svcPort   string
+	logUrl    *url.URL
+	sourceUri *url.URL
+	logType   v1alpha2.LoggerMode
+	modelId   string
+}
+
+func New(log logr.Logger, svcPort string, logUrl *url.URL, sourceUri *url.URL, logType v1alpha2.LoggerMode, modelId string) http.Handler {
+	return &loggerHandler{
+		log:       log,
+		svcPort:   svcPort,
+		logUrl:    logUrl,
+		sourceUri: sourceUri,
+		logType:   logType,
+		modelId:   modelId,
+	}
+}
+
+func (eh *loggerHandler) post(url *url.URL, body []byte, contentType string) ([]byte, error) {
+	eh.log.Info("Calling server", "url", url.String(), "contentType", contentType)
+	response, err := http.Post(url.String(), contentType, bytes.NewReader(body))
+	if err != nil {
+		return nil, err
+	}
+	b, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	if err = response.Body.Close(); err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+func (eh *loggerHandler) callService(b []byte, r *http.Request) ([]byte, string, error) {
```

Comment:
might be cleaner to do *string as the return type. I've been doing this in all cases where there is an error case to avoid awkward returns.